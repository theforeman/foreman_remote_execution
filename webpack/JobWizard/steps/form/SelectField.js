import React, { useState } from 'react';
import { FormGroup, Select, SelectOption } from '@patternfly/react-core';
import PropTypes from 'prop-types';

export const SelectField = ({ label, fieldId, options, value, setValue }) => {
  const onSelect = (event, selection) => {
    setValue(selection);
    setIsOpen(false);
  };
  const [isOpen, setIsOpen] = useState(false);
  return (
    <FormGroup label={label} fieldId={fieldId}>
      <Select
        selections={value}
        onSelect={onSelect}
        onToggle={setIsOpen}
        isOpen={isOpen}
        className="without_select2"
        maxHeight="45vh"
      >
        {options.map((option, index) => (
          <SelectOption key={index} value={option} />
        ))}
      </Select>
    </FormGroup>
  );
};
SelectField.propTypes = {
  label: PropTypes.string.isRequired,
  fieldId: PropTypes.string.isRequired,
  options: PropTypes.array,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  setValue: PropTypes.func.isRequired,
};

SelectField.defaultProps = {
  options: [],
  value: null,
};
