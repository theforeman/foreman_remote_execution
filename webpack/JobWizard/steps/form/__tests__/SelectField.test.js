import React from 'react';
import * as patternfly from '@patternfly/react-core';
import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import { SelectField } from '../SelectField';

jest.spyOn(patternfly, 'Select');
jest.spyOn(patternfly, 'SelectOption');
patternfly.Select.mockImplementation(props => <div>{props}</div>);
patternfly.SelectOption.mockImplementation(props => <div>{props}</div>);
const fixtures = {
  'renders with props': {
    label: 'grouped select',
    fieldId: 'field-id',
    options: ['Commands'],
    value: 'Commands',
    setValue: jest.fn(),
  },
};

describe('SelectField', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(SelectField, fixtures));
});
