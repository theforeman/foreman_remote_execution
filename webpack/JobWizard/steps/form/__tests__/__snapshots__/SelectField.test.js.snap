// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField rendering renders with props 1`] = `
<FormGroup
  fieldId="field-id"
  label="grouped select"
>
  <mockConstructor
    className="without_select2"
    isOpen={false}
    maxHeight="45vh"
    onSelect={[Function]}
    onToggle={[Function]}
    selections="Commands"
  >
    <mockConstructor
      key="0"
      value="Commands"
    />
  </mockConstructor>
</FormGroup>
`;
